api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceAuthService.java

```diff
@@ -99,101 +108,63 @@ public DbWorkspace getWorkspaceEnforceAccessLevelAndSetCdrVersion(
     return workspace;
   }
 
-  // This method has no dependencies so it would be pretty nice if we can define it
-  // on the model itself like so, `currentUpdate.updateFirecloudAclsOnUser(updatedAccess)`
-  // but it's difficult to do that right now since the class is generated by Swagger
-  // Also, the name seems.. inaccurate? It's updating the ACLUpdate object, not a user.
-  public static FirecloudWorkspaceACLUpdate updateFirecloudAclsOnUser(
-      WorkspaceAccessLevel updatedAccess, FirecloudWorkspaceACLUpdate currentUpdate) {
-    if (updatedAccess == WorkspaceAccessLevel.OWNER) {
-      currentUpdate.setCanShare(true);
-      currentUpdate.setCanCompute(true);
-      currentUpdate.setAccessLevel(WorkspaceAccessLevel.OWNER.toString());
-    } else if (updatedAccess == WorkspaceAccessLevel.WRITER) {
-      currentUpdate.setCanShare(false);
-      currentUpdate.setCanCompute(true);
-      currentUpdate.setAccessLevel(WorkspaceAccessLevel.WRITER.toString());
-    } else if (updatedAccess == WorkspaceAccessLevel.READER) {
-      currentUpdate.setCanShare(false);
-      currentUpdate.setCanCompute(false);
-      currentUpdate.setAccessLevel(WorkspaceAccessLevel.READER.toString());
-    } else {
-      currentUpdate.setCanShare(false);
-      currentUpdate.setCanCompute(false);
-      currentUpdate.setAccessLevel(WorkspaceAccessLevel.NO_ACCESS.toString());
-    }
-    return currentUpdate;
-  }
-
   public Map<String, FirecloudWorkspaceAccessEntry> getFirecloudWorkspaceAcls(
       String workspaceNamespace, String firecloudName) {
-    FirecloudWorkspaceACL aclResp =
-        fireCloudService.getWorkspaceAclAsService(workspaceNamespace, firecloudName);
-
-    // Swagger Java codegen does not handle the WorkspaceACL model correctly; it returns a GSON map
-    // instead. Run this through a typed Gson conversion process to parse into the desired type.
-    Type accessEntryType = new TypeToken<Map<String, FirecloudWorkspaceAccessEntry>>() {}.getType();
-    Gson gson = new Gson();
-    return gson.fromJson(gson.toJson(aclResp.getAcl(), accessEntryType), accessEntryType);
```

Comment:
how did you resolve this nonsense?

Reply:
It's still gross, I just centralized it - this was a duplication of a utility method we already had.